src/lib/containers/discussion_forum/DiscussionThread.tsx

```diff
@@ -0,0 +1,171 @@
+import moment from 'moment'
+import React, { useEffect, useState } from 'react'
+import { formatDate } from '../../utils/functions/DateFormatter'
+import {
+  useGetRepliesInfinite,
+  useGetThread,
+} from '../../utils/hooks/SynapseAPI/forum/useForum'
+import { SynapseClient } from '../../utils'
+import { SMALL_USER_CARD } from '../../utils/SynapseConstants'
+import { useSynapseContext } from '../../utils/SynapseContext'
+import {
+  Subscription,
+  SubscriptionObjectType,
+  Topic,
+} from '../../utils/synapseTypes/Subscription'
+import Typography from '../../utils/typography/Typography'
+import { getMessage } from '../DiscussionSearchResult'
+import UserCard from '../UserCard'
+import { getSubscribe } from './ForumTable'
+import { displayToast } from '../ToastMessage'
+import { DiscussionReply } from './DiscussionReply'
+import { Button, FormControl } from 'react-bootstrap'
+import IconSvg from '../IconSvg'
+import Tooltip from '../../utils/tooltip/Tooltip'
+import MarkdownSynapse from '../markdown/MarkdownSynapse'
+import { ObjectType } from '../../utils/synapseTypes'
+
+export type DiscussionThreadProps = {
+  threadId: string
+  limit: number
+}
+
+const FOLLOWING_TEXT = 'You are following this topic. Click to stop following.'
+const UNFOLLOWING_TEXT = 'You are not following this topic. Click to follow.'
+
+export function DiscussionThread(props: DiscussionThreadProps) {
+  const { threadId, limit } = props
+
+  const { accessToken } = useSynapseContext()
+  const [datePosted, setDatePosted] = useState(true)
+  const [message, setMessage] = useState<string>()
+  const [subscribed, setSubscribed] = useState<Subscription>()
```

Comment:
There's a chance to reduce complexity here, and to reduce code duplication since we are fetching and updating a Subscription in multiple components.

Instead of having a state hook, an effect hook, and a complex event handler in this component (and the forum table component), can we extract a custom hook that would look something like this?

```tsx
const { subscription, onToggleSubscription, isLoading } = useSubscription(id, subscriptionObjectType);
```

Then this custom hook could handle the logic for fetching, subscribing, unsubscribing to a particular subscription, and the component just uses those fields that it returns.

It would be especially cool if it used `react-query` 🙂